### DISCUSSION TITLE
.net 8 lambda running in Docker - "Method not found: 'System.String Amazon.Util.Internal.InternalSDKUtils.BuildUserAgentString(System.String, System.String)'." When trying to retrieve secrets

### DISCUSSION THREAD

**Discussion by alex182** [aws/aws-sdk-net#3322] [Q&A] [answered]
Hello, I am attempting to run a .net 8 lambda (not AOT) in a docker container.  Part of the lambda retrieves secrets from secret manager.  The retrieval works fine when running locally in visual studio and using the mock lambda test tool, but once it is deployed in a docker container I get the following exception:

> {
  "errorType": "TypeInitializationException",
  "errorMessage": "The type initializer for 'Amazon.SecretsManager.AmazonSecretsManagerConfig' threw an exception.",
  "stackTrace": [
    "at Amazon.SecretsManager.AmazonSecretsManagerConfig..ctor()",
    "at Amazon.SecretsManager.AmazonSecretsManagerClient..ctor()",
    "at Amazon.SecretsManager.Extensions.Caching.SecretsManagerCache..ctor(SecretCacheConfiguration config)",
    "at Myfunction.Lambda.Providers.SecretsManager.AmazonSecretsManagerConfigurationProvider..ctor() in /home/ubuntu/actions-runner/_work/myfunction/myfunction/src/Myfunction.Lambda/Providers/SecretsManager/AmazonSecretsManagerConfigurationProvider.cs:line 7",
    "at Myfunction.Lambda.Function.GetSecret[T](String secretName) in /home/ubuntu/actions-runner/_work/myfunction/myfunction/src/Myfunction.Lambda/Function.cs:line 287",
    "at Myfunction.Lambda.Function.FunctionHandler(ILambdaContext context) in /home/ubuntu/actions-runner/_work/myfunction/myfunction/src/Myfunction.Lambda/Function.cs:line 46",
    "at lambda_method1(Closure, Stream, ILambdaContext, Stream)",
    "at Amazon.Lambda.RuntimeSupport.HandlerWrapper.<>c__DisplayClass8_0.<GetHandlerWrapper>b__0(InvocationRequest invocation) in /src/Repo/Libraries/src/Amazon.Lambda.RuntimeSupport/Bootstrap/HandlerWrapper.cs:line 54",
    "at Amazon.Lambda.RuntimeSupport.LambdaBootstrap.InvokeOnceAsync(CancellationToken cancellationToken) in /src/Repo/Libraries/src/Amazon.Lambda.RuntimeSupport/Bootstrap/LambdaBootstrap.cs:line 185"
  ],
  "cause":   {
    "errorType": "MissingMethodException",
    "errorMessage": "Method not found: 'System.String Amazon.Util.Internal.InternalSDKUtils.BuildUserAgentString(System.String, System.String)'."
  }
}

**Comment by normj** [comment, marked as answer]
The `BuildUserAgentString(System.String, System.String)` was added back in [December 14, 2023](https://github.com/aws/aws-sdk-net/blob/main/changelogs/SDK.CHANGELOG.2023.md#377090-2023-12-14-1931-utc) as part of version `3.7.300.19` of `AWSSDK.Core`. As part of that release all service packages were updated to call that new method. So if you have new service assemblies but a version of `AWSSDK.Core` that is older then that version you will get this error. 

I would suggest looking in your deployment bundle at the *.deps.json file and see what is causing the older version of the `AWSSDK.Core` being brought into your deployment bundle. If possible you can share the contents of the *.deps.json here for help understanding it.

**Reply by alex182** [reply]
That did it, thank you @normj
